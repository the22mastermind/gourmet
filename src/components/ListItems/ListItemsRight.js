import React from 'react';
import { Caption } from 'react-native-paper';
import { StyleSheet, Text, View } from 'react-native';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-evenly',
    alignItems: 'flex-end',
    marginLeft: 12,
  },
  bottom: {
    fontSize: 14,
    fontWeight: '700',
    color: '#a4c49a',
    textTransform: 'capitalize',
  },
});

const ListItemsRight = ({ top, bottom, ...rest }) => (
  <View style={styles.container} testID="list-item-right-wrapper">
    <Caption testID="list-item-right-top" {...rest}>{top}</Caption>
    <Text testID="list-item-right-bottom" {...rest} style={styles.bottom}>
      {bottom}
    </Text>
  </View>
);

export default ListItemsRight;
