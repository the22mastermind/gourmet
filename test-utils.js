import React from 'react';
import { render } from '@testing-library/react-native';
import PaperProvider from 'react-native-paper/lib/commonjs/core/Provider';

const AllTheProviders = ({ children }) => (
  <PaperProvider>{children}</PaperProvider>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
