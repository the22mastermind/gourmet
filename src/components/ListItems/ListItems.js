import React from 'react';
import { StyleSheet } from 'react-native';
import { Avatar, List } from 'react-native-paper';
import PropTypes from 'prop-types';
import ListItemsRight from './ListItemsRight';

const styles = StyleSheet.create({
  description: {
    marginTop: 6,
  },
});

const ListItems = ({ items, handleItem }) => (
  items.map(item => (
    <List.Item
      key={item.id}
      testID={item?.name || item?.paymentId}
      title={item?.name || item?.paymentId}
      description={item?.description || `${item?.Contents.length} items`}
      descriptionStyle={styles.description}
      left={props => item?.image ? (
        <Avatar.Image
          {...props}
          testID="item-image"
          source={{
            uri: item?.image,
            cache: 'force-cache'
          }}
        />
      ) : null}
      right={props => (
        <ListItemsRight
          top={item?.size ? item.size : item.createdAt.toLocaleString()}
          bottom={item?.cost ? `$${item.cost}` : item.status}
          {...props}
        />
      )}
      onPress={() => handleItem(item)}
  />
  ))
);

ListItems.propTypes = {
  items: PropTypes.arrayOf(Object).isRequired,
  handleItem: PropTypes.func.isRequired,
};

export default ListItems;
