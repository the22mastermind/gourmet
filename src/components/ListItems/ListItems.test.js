import React from 'react';
import { render, fireEvent } from 'test-utils';
import ListItems from './ListItems';

const menu = {
  items: [
    {
      menuId: 2,
      id: 2,
      name: 'Double Cheese Burger',
      description: 'This is a very tasty cheese burger.',
      cost: 6.50,
      size: 'Large',
      image: 'https://media.istockphoto.com/photos/delicious-fresh-cooked-burger-with-a-side-of-french-fries-picture-id177556385',
    },
  ],
  handleItem: jest.fn(),
};

describe('<ListItems /> component', () => {
  it('renders a menu <ListItems /> component correctly', () => {
    const screen = render(<ListItems {...menu} />);
  
    expect(screen.getByTestId(menu.items[0].name)).toBeDefined();
    expect(screen.queryByText('Double Cheese Burger')).toBeDefined();
    expect(screen.queryByText('This is a very tasty cheese burger.')).toBeDefined();
    expect(screen.getByTestId('item-image')).toBeDefined();
    expect(screen.getByTestId('list-item-right-wrapper')).toBeDefined();
    expect(screen.getByTestId('list-item-right-top').props.children).toBe('Large');
    expect(screen.getByTestId('list-item-right-bottom').props.children).toBe('$6.5');
  });
  
  it('calls the onPress event correctly', async () => {
    const screen = render(<ListItems {...menu} />);

    fireEvent.press(screen.getByTestId(menu.items[0].name));
  
    expect(menu.handleItem).toHaveBeenCalledWith(menu.items[0]);
  });
  
  it('matches <ListItems /> component snapshot', async () => {
    const screen = render(<ListItems {...menu} />);
  
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
