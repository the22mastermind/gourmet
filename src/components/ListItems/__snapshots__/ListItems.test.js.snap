// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItems /> component matches <ListItems /> component snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
    testID="2"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        color="rgba(51, 51, 51, 0.54)"
        style={
          Array [
            Object {
              "backgroundColor": "#a4c49a",
              "borderRadius": 32,
              "height": 64,
              "width": 64,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 16,
            },
          ]
        }
        testID="item-image"
      >
        <Image
          source={
            Object {
              "cache": "force-cache",
              "uri": "https://media.istockphoto.com/photos/delicious-fresh-cooked-burger-with-a-side-of-french-fries-picture-id177556385",
            }
          }
          style={
            Object {
              "borderRadius": 32,
              "height": 64,
              "width": 64,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(51, 51, 51, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          Double Cheese Burger
        </Text>
        <Text
          numberOfLines={2}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 14,
                },
                Object {
                  "color": "rgba(51, 51, 51, 0.54)",
                },
                Object {
                  "marginTop": 6,
                },
              ],
            ]
          }
        >
          This is a very tasty cheese burger.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "justifyContent": "space-evenly",
            "marginLeft": 12,
          }
        }
        testID="list-item-right-wrapper"
      >
        <Text
          color="rgba(51, 51, 51, 0.54)"
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "rgba(51, 51, 51, 0.54)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Object {
                    "marginRight": 0,
                  },
                ],
              ],
            ]
          }
          testID="list-item-right-top"
        >
          Large
        </Text>
        <Text
          color="rgba(51, 51, 51, 0.54)"
          style={
            Object {
              "color": "#a4c49a",
              "fontSize": 14,
              "fontWeight": "700",
              "textTransform": "capitalize",
            }
          }
          testID="list-item-right-bottom"
        >
          $6.5
        </Text>
      </View>
    </View>
  </View>
</View>
`;
